pkg/rtc/participant.go

```diff
@@ -470,27 +436,9 @@ func (p *ParticipantImpl) ICERestart() error {
 	})
 }
 
-// AddSubscriber subscribes op to all publishedTracks
-func (p *ParticipantImpl) AddSubscriber(op types.Participant) (int, error) {
-	tracks := p.GetPublishedTracks()
-
-	if len(tracks) == 0 {
-		return 0, nil
-	}
-
-	p.params.Logger.Debugw("subscribing new participant to tracks",
-		"subscriber", op.Identity(),
-		"subscriberID", op.ID(),
-		"numTracks", len(tracks))
-
-	n := 0
-	for _, track := range tracks {
-		if err := track.AddSubscriber(op); err != nil {
-			return n, err
-		}
-		n += 1
-	}
-	return n, nil
+// AddSubscriber subscribes op to all publishedTracks or given set of tracks
+func (p *ParticipantImpl) AddSubscriber(op types.Participant, params types.AddSubscriberParams) (int, error) {
```

Comment:
Change in `Participant` interface method to take params to specify adding all tracks or specific tracks.